### PR TITLE
Boos 318/feature/dev app

### PR DESCRIPTION
**Pull request by koutsilisk**
in this PR we introduce:
- Python manage file to delete/recreate/seed db from terminal. 
- Clean and better setup on config file.
- Option to start at with dev or prod config depending on FLASK_DEBUG env variable.

### CODE REVIEW

File: backend/README.md [hunk 1]
```diff
@@ -24,18 +32,34 @@
...
+   python manage.py seed_db
+   ```
+
+8. Run app
    ```bash
    flask run
    ```
-   The app is running at localhost:5000
\ No newline at end of file
+   The app is running at localhost:5000
```
Comment (Chichilele) [inline backend/README.md:65]: the API not the app

File: backend/config.py [hunk 2]
```diff
@@ -50,27 +50,48 @@ class BaseConfig(object):
...
 class ProdConfig(BaseConfig):
     """Production configuration."""
 
-    ENV = "prod"
+    # ENV and DEBUG may not work with flask run
+    # and not recomended to set in code:
+    # https://flask.palletsprojects.com/en/1.1.x/config/#environment-and-debug-features
+    # should be set from terminal
+    # left for info purposes
+
```
Comment (Chichilele) [inline backend/config.py:64]: Simpler
```python
    # ENV and DEBUG should be set in the terminal, not in code. Left for info purposes:
    # https://flask.palletsprojects.com/en/1.1.x/config/#environment-and-debug-features
```

File: backend/config.py [hunk 3]
```diff
@@ -50,27 +50,48 @@ class BaseConfig(object):
...
+    ENV = "production"
     DEBUG = False
 
+    # Password hash
+    BCRYPT_LOG_ROUNDS = 14
+
+    SQLALCHEMY_DATABASE_URI = os.environ.get(
+        "DATABASE_URL", "sqlite:///" + os.path.join(basedir, "prod.db")
+    )  # TODO migrate to postgresql
```
Comment (Chichilele) [inline backend/config.py:73]: Remove comment: `BaseConfig` should be sqlite, not SQL

File: backend/config.py [hunk 4]
```diff
@@ -50,27 +50,48 @@ class BaseConfig(object):
...
     MAIL_BACKEND = "smtp"
+    MAIL_SERVER = os.environ.get("FLASK_MAIL_SERVER", "smtp.gmail.com")
+    MAIL_PORT = int(os.environ.get("FLASK_MAIL_PORT", 587))  # 465 for SSL, 587 for TLS
 
 
 class DevConfig(BaseConfig):
     """Development configuration."""
 
-    ENV = "dev"
+    ENV = "development"
```
Comment (Chichilele) [inline backend/config.py:84]: Same comment about ENV and DEBUG

**Review by Chichilele — APPROVED** [review]

File: backend/README.md [hunk 5]
```diff
@@ -1,6 +1,14 @@
-
 # SPA-Flask-React-boilerplate API
 
+## Development config
+
+To start app with development config you have to set FLASK_DEBUG env variable.
+
+```bash
+   export FLASK_DEBUG=True
+   export FLASK_ENV=development # optional
```
[the remark this answers is not in the document]
Comment (koutsilisk) [reply backend/README.md:9]: yes, but I think is better to set the debug to be sure and the env as optional, if you don't set it it wont change something if the debug is set

File: backend/config.py (hunk 3 above)
Comment (koutsilisk) [reply backend/config.py:73]: that's ProdConfig

**Pull request merged**